network-primitives/scripts/create_domain_node_compose_file.sh

```diff
@@ -0,0 +1,168 @@
+#!/bin/bash
+
+cat > /subspace/docker-compose.yml << EOF
+version: "3.7"
+
+volumes:
+  caddy_data: {}
+  archival_node_data: {}
+
+services:
+  datadog:
+    container_name: "datadog_agent"
+    image: gcr.io/datadoghq/agent:7
+    restart: unless-stopped
+    environment:
+      - DD_API_KEY=\$DATADOG_API_KEY
+      - DD_SITE=datadoghq.com
+    volumes:
+      - /var/run/docker.sock:/var/run/docker.sock:ro
+      - /proc/:/host/proc/:ro
+      - /sys/fs/cgroup/:/host/sys/fs/cgroup:ro
+      - /etc/os-release:/host/etc/os-release:ro
+
+  # caddy reverse proxy with automatic tls management using let encrypt
+  caddy:
+    image: lucaslorentz/caddy-docker-proxy:latest
+    restart: unless-stopped
+    ports:
+      - "80:80"
+      - "443:443"
+    environment:
+      - CADDY_INGRESS_NETWORKS=subspace_default
+    volumes:
+      - /var/run/docker.sock:/var/run/docker.sock:ro
+      - caddy_data:/data
+
+  archival-node:
+    image: ghcr.io/\${NODE_ORG}/node:\${NODE_TAG}
+    volumes:
+      - archival_node_data:/var/subspace:rw
+    restart: unless-stopped
+    ports:
+      - "30333:30333"
+      - "\${NODE_DSN_PORT}:30433"
+    labels:
+      caddy_0: \${DOMAIN_PREFIX}-\${NODE_ID}.\${NETWORK_NAME}.subspace.network

```

Comment:
i dont think we need to expose the primary rpc here. It is already exposed under rpc node anyway